lib/services/saml.go

```diff
@@ -161,11 +161,20 @@ func GetSAMLServiceProvider(sc SAMLConnector, clock clockwork.Clock) (*saml2.SAM
 		return nil, trace.BadParameter("no identity provider certificate provided, either set certificate as a parameter or via entity_descriptor")
 	}
 
-	keyStore, err := utils.ParseSigningKeyStorePEM(sc.GetSigningKeyPair().PrivateKey, sc.GetSigningKeyPair().Cert)
+	signingKeyStore, err := utils.ParseSigningKeyStorePEM(sc.GetSigningKeyPair().PrivateKey, sc.GetSigningKeyPair().Cert)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	encryptionKeyStore := signingKeyStore
```

Comment:
This is surprising behavior, why did you default to using the signing keys for encryption?

Reply:
That's what gosaml2 does. If encryptionkeystore is nil it will use spsigningkeystore instead.

Reply:
I simply defaulted the keystore to the signing one here but it could just as well be nil and we'd get the exact same behaviour.

Reply:
That makes sense, can you add a comment explaining that.